pkg/operator/ceph/cluster/controller.go

```diff
@@ -263,6 +323,61 @@ func (c *ClusterController) onAdd(obj interface{}) {
 // ************************************************************************************************
 // Update event functions
 // ************************************************************************************************
+func (c *ClusterController) onK8sNodeUpdate(oldObj, newObj interface{}) {
+	// skip forced resyncs
+	if reflect.DeepEqual(oldObj, newObj) {
+		return
+	}
+
+	// Checking for nodes where NoSchedule-Taint got removed
+	newNode, ok := newObj.(*v1.Node)
+	if !ok {
+		logger.Warningf("Expected NodeList but handler received %#v", newObj)
+	}
+
+	oldNode, ok := oldObj.(*v1.Node)
+	if !ok {
+		logger.Warningf("Expected NodeList but handler received %#v", oldObj)
+	}
+
+	for i := range newNode.Spec.Taints {
+		if newNode.Spec.Taints[i].Effect == "NoSchedule" {
+			logger.Debugf("Skipping -> Node %s is Unschedulable", newNode.Labels[apis.LabelHostname])
+			return
+		}
+	}
+	var found = false
+	for i := range oldNode.Spec.Taints {
+		if oldNode.Spec.Taints[i].Effect == "NoSchedule" {
+			found = true
+			break
+		}
+	}
+
+	if found == false {
+		logger.Debugf("Skipping -> Node %s was Schedulable anyhow", oldNode.Labels[apis.LabelHostname])
+		return
+	}
+
+	for _, cluster := range c.clusterMap {
+		if cluster.Spec.Storage.AnyUseAllDevices() == false {
```

Comment:
We shouldn't need to skip the node if useAllDevices is false. There could be other device settings that are valid for the new node.